ee/query-service/app/api/api.go

```diff
@@ -0,0 +1,124 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"go.signoz.io/signoz/ee/query-service/dao"
+	"go.signoz.io/signoz/ee/query-service/interfaces"
+	"go.signoz.io/signoz/ee/query-service/license"
+	baseapp "go.signoz.io/signoz/pkg/query-service/app"
+	baseint "go.signoz.io/signoz/pkg/query-service/interfaces"
+	rules "go.signoz.io/signoz/pkg/query-service/rules"
+	"go.signoz.io/signoz/pkg/query-service/version"
+)
+
+type APIHandlerOptions struct {
+	DataConnector  interfaces.DataConnector
+	AppDao         dao.ModelDao
+	RulesManager   *rules.Manager
+	FeatureFlags   baseint.FeatureLookup
+	LicenseManager *license.Manager
+}
+
+type APIHandler struct {
+	opts APIHandlerOptions
+	baseapp.APIHandler
+}
+
+// NewAPIHandler returns an APIHandler
+func NewAPIHandler(opts APIHandlerOptions) (*APIHandler, error) {
+
+	baseHandler, err := baseapp.NewAPIHandler(baseapp.APIHandlerOpts{
+		Reader:       opts.DataConnector,
+		AppDao:       opts.AppDao,
+		RuleManager:  opts.RulesManager,
+		FeatureFlags: opts.FeatureFlags})
+
+	if err != nil {
+		return nil, err
+	}
+
+	ah := &APIHandler{
+		opts:       opts,
+		APIHandler: *baseHandler,
+	}
+	return ah, nil
+}
+
+func (ah *APIHandler) FF() baseint.FeatureLookup {
+	return ah.opts.FeatureFlags
+}
+
+func (ah *APIHandler) RM() *rules.Manager {
+	return ah.opts.RulesManager
+}
+
+func (ah *APIHandler) LM() *license.Manager {
+	return ah.opts.LicenseManager
+}
+
+func (ah *APIHandler) AppDao() dao.ModelDao {
+	return ah.opts.AppDao
+}
+
+func (ah *APIHandler) CheckFeature(f string) bool {
+	err := ah.FF().CheckFeature(f)
```

Comment:
Mentioned this in prior comment. The intent was to not copy the references from Opts into the struct var (e..g APIHandler) but some code in the base (oss API handler) is still copying the reference from Opts.  

The opts would carry required references (pointers) and can continue to serve them. 
This means instead of 
```

NewAPIHandler(opts APIHandlerOpts) *APIHandler{
return APIHandler {
  featureFlags: opts.FeatureFlags,
  ...  
 }
}
```

we would carry opts as it is to reduce this copying.

NewAPIHandler(opts APIHandlerOpts) *APIHandler{
return APIHandler {
  opts: opts,
  ...  
 }
}

Reply:
It will be opts. I didnt want to make much change in the base version (non-ee) in this PR. If you look at EE, it already just uses opts. 
I can submit a separate for base version (non-ee)